### PR TITLE
[v6r17] allow to run multiple DataStore

### PR DESCRIPTION
**Pull request by zmathe**
In case when one DataStore can not scope with the load, we can install multiple helper DataStore services. These services are only inserting the records to the appropriate table. The master service will do the bucketing.

### CODE REVIEW

**Comment by chaen** [comment]
Can you just update the doc, so it is done once and for all ? We need to specify the constraint that only one DataStore can run the bucketing, and how to define the URL. Thanks

**Comment by zmathe** [comment]
Done

File: AccountingSystem/Service/DataStoreHandler.py [hunk 1]
```diff
@@ -21,11 +22,14 @@ class DataStoreHandler( RequestHandler ):
...
-    cls.__acDB.autoCompactDB() #pylint: disable=no-member
-    result = cls.__acDB.markAllPendingRecordsAsNotTaken() #pylint: disable=no-member
-    if not result[ 'OK' ]:
-      return result
-    gThreadScheduler.addPeriodicTask( 60, cls.__acDB.loadPendingRecords ) #pylint: disable=no-member
+    #we can run multiple services in read only mode. In that case we do not bucket
+    cls.runBucketing = getServiceOption( svcInfoDict, 'RunBucketing', True )
+    if cls.runBucketing:
+      cls.__acDB.autoCompactDB() #pylint: disable=no-member
+      result = cls.__acDB.markAllPendingRecordsAsNotTaken() #pylint: disable=no-member
```
Comment (fstagni) [inline AccountingSystem/Service/DataStoreHandler.py]: Isn't only `cls.__acDB.autoCompactDB()` that is needed for the bucketing?

File: AccountingSystem/Service/DataStoreHandler.py [hunk 2]
```diff
@@ -146,7 +150,10 @@ def export_compactDB( self ):
     """
     Compact the db by grouping buckets
     """
-    return self.__acDB.compactBuckets() #pylint: disable=no-member
+    if self.runBucketing:
+      return self.__acDB.compactBuckets() #pylint: disable=no-member
+    else:
+      return RPCClient('Accounting/DataStoreMaster').compactDB()
```
Comment (fstagni) [inline AccountingSystem/Service/DataStoreHandler.py]: Please put a note to refer to documentation here (this is not necessarily a fixed name)

File: docs/source/AdministratorGuide/Systems/Accounting/index.rst [hunk 3]
```diff
@@ -39,3 +39,22 @@ Since v6r12 each accounting type can be stored in a different DB. By default all
     }
     
 With the previous configuration all accounting data will be stored and retrieved from the usual database except for the _WMSHistory_ type that will be stored and retrieved from the _Acc2_ database.
+
+DataStore Helpers
+======================
+From DIRAC v8r17p14 we are able to run multiple services. The master will creates the proper buckets and the helpers only insert the records to the 'in' table.
```
Comment (fstagni) [inline docs/source/AdministratorGuide/Systems/Accounting/index.rst]: DIRAC v6r17p4

Comment (fstagni) [inline docs/source/AdministratorGuide/Systems/Accounting/index.rst]: I'd rephrase to 

"From DIRAC v6r17p14 there is the possibility to to run multiple 'DataStore' services, where one need to be called 'master', while all the others may be called slaves. The master will creates the proper buckets and the helpers only insert the records to the 'in' table."

File: AccountingSystem/Service/DataStoreHandler.py (hunk 1 above)
Comment (zmathe) [inline AccountingSystem/Service/DataStoreHandler.py]: yes, autoCompactDB does the work! but we do not want to change the status of pending requests also, if one of the service restart....

File: AccountingSystem/Service/DataStoreHandler.py (hunk 2 above)
Comment (zmathe) [inline AccountingSystem/Service/DataStoreHandler.py]: ok

File: docs/source/AdministratorGuide/Systems/Accounting/index.rst (hunk 3 above)
Comment (zmathe) [inline docs/source/AdministratorGuide/Systems/Accounting/index.rst]: replaced
